File: LeetCode/C#/700-799/739.cs
using System;
using System.Collections.Generic;

namespace LeetCode739
{
    public class Solution {
        public int[] DailyTemperatures(int[] T) {
            Stack<int> st = new Stack<int>();
            int size = T.Length;

            int[] ans = new int[size];

            for(int i = 0 ; i < size;++i){
                while(st.Count > 0){
                    if(T[i] > T[st.Peek()]){
                        ans[st.Peek()] = i - st.Peek();
                        st.Pop();
                    }
                    else{
                        break;
                    }
                }

                st.Push(i);
            }
        
            return ans;
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            int[] t = new int[] {73, 74, 75, 71, 69, 72, 76, 73};
            int[] ans = new Solution().DailyTemperatures(t);
            
            foreach (int item in ans)
            {
                Console.WriteLine(item);
            }

            Console.ReadLine();
        }
    }
}

File: LeetCode/C++/700-799/739.cpp
#include <iostream>
#include <stack>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& T) {
        stack<int> st;
        int size = T.size();
        vector<int> ans = vector<int>(size, 0);

        for(int i = 0 ; i < size;++i){
            while(!st.empty()){
                if(T[i] > T[st.top()]){
                    ans[st.top()] = i - st.top();
                    st.pop();
                }
                else{
                    break;
                }
            }

            st.push(i);
        }
        
        return ans;
    }
};


int main()
{
	vector<int> T{73, 74, 75, 71, 69, 72, 76, 73}; 
	Solution sol;
	vector<int> ans = sol.dailyTemperatures(T);
    
	for(int i = 0 ; i < ans.size();++i){
		cout << ans[i] << endl;
	}

    return 0;
}
